src/sagemaker/tuner.py

```diff
@@ -1580,6 +1583,9 @@ def _prepare_training_config(
         if parameter_ranges is not None:
             training_config["parameter_ranges"] = parameter_ranges
 
+        if estimator.max_retry_attempts is not None:
+            training_config["max_retry_attempts"] = estimator.max_retry_attempts
```

Comment:
do we already have a standard mechanism to propagate API changes to pipeline steps? i.e. simultaneous release of new API features and pipeline support

Reply:
from DSL level yes, as long as the model has it, we can support it simultaneously. However, in python SDK, when translating the estimator or tuner to request arguments, one will need to manually add the new fields. For instance, in this case, the estimator supports `max_retry_attempts`, however when passed to tuner, and during python object to request arguments translations, this is not included.